common/src/View/MapFrame.cpp

```diff
@@ -130,8 +131,29 @@ namespace TrenchBroom {
             m_updateLocker->Start();
 #endif
         }
+        
+        static RenderView* FindChildRenderView(wxWindow *current) {
+            for (wxWindow *child : current->GetChildren()) {
+                RenderView *canvas = dynamic_cast<RenderView*>(child);
+                if (canvas)
+                    return canvas;
+                
+                canvas = FindChildRenderView(child);
+                if (canvas)
+                    return canvas;
+            }
+            return nullptr;
+        }
 
         MapFrame::~MapFrame() {
+            // Search for a RenderView (wxGLCanvas subclass) and make it current.
+            RenderView* canvas = FindChildRenderView(this);
+            if (canvas != nullptr && m_contextManager != nullptr) {
+                wxGLContext* mainContext = m_contextManager->mainContext();
+                if (mainContext != nullptr)
+                    mainContext->SetCurrent(*canvas);
+            }
+            
             // Makes IsBeingDeleted() return true
             SendDestroyEvent();
```

Comment:
No, the docs say wxWindowDestroyEvent is normally sent by the wxFrame destructor, but you can call `SendDestroyEvent()` to send it in your subclass's destructor if needed:
http://docs.wxwidgets.org/3.1.0/classwx_window_destroy_event.html